tests/ui/lifetimes/temporary-lifetime-extension.rs

```diff
@@ -0,0 +1,29 @@
+//@ check-pass
+
+fn temp() -> (String, i32) {
+    (String::from("Hello"), 1)
+}
+
+fn main() {
+    let a = &temp();
+    let b = [(&temp(),)];
+    let c = &temp().0;
+    let d = &temp().0[..];
+    let e = {
+        let _ = 123;
+        &(*temp().0)[..]
+    };
+    let f = if true {
+        &temp()
+    } else {
+        &temp()
+    };
+    let g = match true {
+        true => &temp(),
+        false => {
+            let _ = 123;
+            &temp()
+        }
+    };
+    println!("{a:?} {b:?} {c:?} {d:?} {e:?} {f:?} {g:?}");
```

Comment:
I think the following (with, _e.g._, `::std::env::vars().nth(1)`), would be useful to ensure as well:

```suggestion
    };
    let h = match temp() {
        owned_non_temporary => &{ owned_non_temporary },
    };
    println!("{a:?} {b:?} {c:?} {d:?} {e:?} {f:?} {g:?} {h:?}");
```